### PR TITLE
feat(read-replica): Add database config for postgres read replicas

### PR DESCRIPTION
**Pull request by ellie**
## Problem

Allow for configuring a second postgres database, this time a
read-replica.

This is configured via the POSTHOG_POSTGRES_READ_HOST, introduced here: https://github.com/PostHog/charts-clickhouse/pull/722

Note that with our current setup, all replicas have the same port/db
name/creds etc - only the host differs.

This change will not actually change any behaviour until the following
are complete

1. Write a new database router to actually use this
2. Set the env var to add the config

I will follow up with those :)

## How did you test this code?
I couldn't find any testing for our settings stuff, so did not add to it + think starting one now would be a bit out of scope. We probably should though.

### CODE REVIEW

**Review by neilkakkar — APPROVED** [review]

File: posthog/settings/data_stores.py [hunk 1]
```diff
@@ -78,6 +95,12 @@
         f'The environment vars "DATABASE_URL" or "POSTHOG_DB_NAME" are absolutely required to run this software'
     )
 
+# Configure the database which will be used as a read replica.
+# This should have all the same config as our main writer DB, just use a different host.
+# Our database router will point here.
+if os.getenv("POSTHOG_POSTGRES_READ_HOST"):
+    DATABASES["replica"] = postgres_config(os.getenv("POSTHOG_POSTGRES_READ_HOST", "localhost"))
```
[the remark this answers is not in the document]
Comment (ellie) [reply posthog/settings/data_stores.py:102]: True! Serves me right for copy-pasting. Thanks

File: posthog/settings/data_stores.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import os
+from typing import Dict, Any
```
Comment (tomasfarias) [inline posthog/settings/data_stores.py:2]: nit: In Python 3.10 you can just use `dict`.

Comment (ellie) [reply posthog/settings/data_stores.py:2]: oh lovely! thanks 🤩

**Review by tomasfarias — APPROVED** [review]
Neat 🚀🚀🚀 Looking forward to more replicas.

**Pull request merged**